ui/core/proto_utils/equipped_item.ts

```diff
@@ -293,6 +402,16 @@ export class EquippedItem {
 		return !!this._item.randomSuffixOptions.length;
 	}
 
+	hasUpgradeOptions() {
+		const { scalingOptions } = this.item;
+		// Make sure to always exclude Challenge Mode scaling options as those are handled globally
+		// and offset these options by 1 due to items always having a base option.
+		delete scalingOptions[ItemLevelState.ChallengeMode];
+
+		return Object.values(scalingOptions);
+
+	}
```

Comment:
This method is worded as if it will return a `boolean`, but actually returns an array. I would also recommend making the return type explicit in the function signature.